### ISSUE TITLE
Consul Service Discovery: add support for health API endpoint

### ISSUE THREAD

**Issue by chen23** [F5Networks/f5-appsvcs-extension#163]
##### ISSUE TYPE

 - Feature Idea

##### AS3 BUILD/ VERSION

```
{
    "version": "3.14.0",
    "release": "4",
    "schemaCurrent": "3.14.0",
    "schemaMinimum": "3.0.0"
}
```

##### BIGIP VERSION

```

Sys::Version
Main Package
  Product     BIG-IP
  Version     13.1.1.2
  Build       0.0.4
  Edition     Point Release 2
  Date        Thu Oct 11 15:32:21 PDT 2018
```

##### SUMMARY

Currently Consul SD expects either a "node" or "catalog" output of addresses.  This does not indicate whether a service is healthy or not.  To match behavior of consul DNS it should be able to handle output from the /v1/health API as well.

##### STEPS TO REPRODUCE

catalog output

/v1/catalog/service/pool001
```
[
  {
    "ID": "",
    "Node": "192.168.128.1",
    "Address": "192.168.128.1",
    "Datacenter": "dc1",
    "TaggedAddresses": null,
    "NodeMeta": {
      "external-node": "true",
      "external-probe": "false"
    },
    "ServiceKind": "",
    "ServiceID": "Production-pool001-1",
    "ServiceName": "pool001",
    "ServiceTags": [
      "as3",
      "Production"
    ],
    "ServiceAddress": "",
    "ServiceWeights": {
      "Passing": 1,
      "Warning": 1
    },
    "ServiceMeta": {},
    "ServicePort": 443,
    "ServiceEnableTagOverride": false,
    "ServiceProxy": {
      "MeshGateway": {}
    },
    "ServiceConnect": {},
    "CreateIndex": 71477,
    "ModifyIndex": 71477
  },
  {
    "ID": "",
    "Node": "192.168.160.1",
    "Address": "192.168.160.1",
    "Datacenter": "dc1",
    "TaggedAddresses": null,
    "NodeMeta": {
      "external-node": "true",
      "external-probe": "false"
    },
    "ServiceKind": "",
    "ServiceID": "Production-pool001-2",
    "ServiceName": "pool001",
    "ServiceTags": [
      "as3",
      "Production"
    ],
    "ServiceAddress": "",
    "ServiceWeights": {
      "Passing": 1,
      "Warning": 1
    },
    "ServiceMeta": {},
    "ServicePort": 443,
    "ServiceEnableTagOverride": false,
    "ServiceProxy": {
      "MeshGateway": {}
    },
    "ServiceConnect": {},
    "CreateIndex": 71667,
    "ModifyIndex": 71667
  },
  {
    "ID": "",
    "Node": "192.168.192.1",
    "Address": "192.168.192.1",
    "Datacenter": "dc1",
    "TaggedAddresses": null,
    "NodeMeta": {
      "external-node": "true",
      "external-probe": "false"
    },
    "ServiceKind": "",
    "ServiceID": "Production-pool001-3",
    "ServiceName": "pool001",
    "ServiceTags": [
      "as3",
      "Production"
    ],
    "ServiceAddress": "",
    "ServiceWeights": {
      "Passing": 1,
      "Warning": 1
    },
    "ServiceMeta": {},
    "ServicePort": 443,
    "ServiceEnableTagOverride": false,
    "ServiceProxy": {
      "MeshGateway": {}
    },
    "ServiceConnect": {},
    "CreateIndex": 71670,
    "ModifyIndex": 71670
  }
]
```
health output (does not handle this)

/v1/health/service/pool001?passing

```[
  {
    "Node": {
      "ID": "",
      "Node": "192.168.192.1",
      "Address": "192.168.192.1",
      "Datacenter": "dc1",
      "TaggedAddresses": null,
      "Meta": {
        "external-node": "true",
        "external-probe": "false"
      },
      "CreateIndex": 71670,
      "ModifyIndex": 71670
    },
    "Service": {
      "ID": "Production-pool001-3",
      "Service": "pool001",
      "Tags": [
        "as3",
        "Production"
      ],
      "Address": "",
      "Meta": {},
      "Port": 443,
      "Weights": {
        "Passing": 1,
        "Warning": 1
      },
      "EnableTagOverride": false,
      "Proxy": {
        "MeshGateway": {}
      },
      "Connect": {},
      "CreateIndex": 71670,
      "ModifyIndex": 71670
    },
    "Checks": [
      {
        "Node": "192.168.192.1",
        "CheckID": "http-check",
        "Name": "http-check",
        "Status": "passing",
        "Notes": "",
        "Output": "HTTP GET https://192.168.192.1: 200 OK Output: ",
        "ServiceID": "",
        "ServiceName": "",
        "ServiceTags": [],
        "Definition": {
          "Interval": "5s",
          "HTTP": "https://192.168.192.1",
          "TLSSkipVerify": true,
          "Header": {
            "Host": [
              "pool001.example.com"
            ],
            "x-monitored-by": [
              "eric"
            ]
          },
          "Method": "HEAD"
        },
        "CreateIndex": 71670,
        "ModifyIndex": 71726
      }
    ]
  },
  {
    "Node": {
      "ID": "",
      "Node": "192.168.160.1",
      "Address": "192.168.160.1",
      "Datacenter": "dc1",
      "TaggedAddresses": null,
      "Meta": {
        "external-node": "true",
        "external-probe": "false"
      },
      "CreateIndex": 71667,
      "ModifyIndex": 71667
    },
    "Service": {
      "ID": "Production-pool001-2",
      "Service": "pool001",
      "Tags": [
        "as3",
        "Production"
      ],
      "Address": "",
      "Meta": {},
      "Port": 443,
      "Weights": {
        "Passing": 1,
        "Warning": 1
      },
      "EnableTagOverride": false,
      "Proxy": {
        "MeshGateway": {}
      },
      "Connect": {},
      "CreateIndex": 71667,
      "ModifyIndex": 71667
    },
    "Checks": [
      {
        "Node": "192.168.160.1",
        "CheckID": "http-check",
        "Name": "http-check",
        "Status": "passing",
        "Notes": "",
        "Output": "HTTP GET https://192.168.160.1: 200 OK Output: ",
        "ServiceID": "",
        "ServiceName": "",
        "ServiceTags": [],
        "Definition": {
          "Interval": "5s",
          "HTTP": "https://192.168.160.1",
          "TLSSkipVerify": true,
          "Header": {
            "Host": [
              "pool001.example.com"
            ],
            "x-monitored-by": [
              "eric"
            ]
          },
          "Method": "HEAD"
        },
        "CreateIndex": 71667,
        "ModifyIndex": 71723
      }
    ]
  }
]
```

##### EXPECTED RESULTS

ability to query health API
##### ACTUAL RESULTS

unable to query health API

**Comment by dstokesf5** [comment]
Could you set up a BIG-IP health monitor to check the health API? Would adding better support for monitors to service discovery nodes properly cover your use case here?

**Comment by chen23** [comment]
The use case is to directly use Consul's health API.  I do not know of anyone else asking for this, but folks are welcome to +1.

**Comment by chrisjohnson** [comment]
We would love to see this, and agree that the point is to let consul handle the health checking. We would like to reduce and eventually maybe even remove the F5 health checks and let consul inform F5 about the health

**Comment by dstokesf5** [comment]
This will be resolved by the solution for AUTOTOOL-576 (#117).

**Comment by chen23** [comment]
This has been resolved by AS3 3.24
